### PR TITLE
Add frontend support for `time` platform

### PR DESCRIPTION
**Pull request by raman325**
## Proposed change
Adds support for the `time` platform outlined in https://github.com/home-assistant/architecture/discussions/811

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- Link to core pull request: https://github.com/home-assistant/core/pull/81949
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/24987

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by raman325** [comment]
not stale

File: src/common/entity/compute_state_display.ts [hunk 1]
```diff
@@ -130,7 +130,7 @@ export const computeStateDisplayFromEntityAttributes = (
         date = new Date(attributes.year, attributes.month - 1, attributes.day);
         return formatDate(date, locale);
       }
-      if (attributes.has_time) {
+      if (domain === "time" || attributes.has_time) {
         date = new Date();
         date.setHours(attributes.hour, attributes.minute);
```
Comment (frenck) [inline src/common/entity/compute_state_display.ts:135]: The `time` entity doesn't have these attributes. The above time parsing should already cover it, right?

File: src/data/time.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  HassEntityAttributeBase,
+  HassEntityBase,
+} from "home-assistant-js-websocket";
+import { HomeAssistant } from "../types";
+
+interface TimeEntityAttributes extends HassEntityAttributeBase {
+  hour?: number;
+  minute?: number;
+  second?: number;
+}
```
Comment (frenck) [inline src/data/time.ts:11]: These attributes have been removed

**Comment by frenck** [comment]
Backend has been merged.

**Review by bramkragten — APPROVED** [review]

File: src/panels/lovelace/entity-rows/hui-time-entity-row.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  static get styles(): CSSResultGroup {
+    return css`
+      ha-date-input + ha-time-input {
+        margin-left: 4px;
+        margin-inline-start: 4px;
+        margin-inline-end: initial;
+        direction: var(--direction);
+      }
+    `;
+  }
```
[the remark this answers is not in the document]
Comment (raman325) [reply src/panels/lovelace/entity-rows/hui-time-entity-row.ts:88]: https://github.com/home-assistant/frontend/pull/16416

**Pull request merged**